source/extensions/filters/network/mysql_proxy/mysql_codec_clogin.cc

```diff
@@ -0,0 +1,160 @@
+#include "extensions/filters/network/mysql_proxy/mysql_codec_clogin.h"
+
+#include "extensions/filters/network/mysql_proxy/mysql_codec.h"
+#include "extensions/filters/network/mysql_proxy/mysql_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace MySQLProxy {
+
+void ClientLogin::setClientCap(int client_cap) { client_cap_ = client_cap; }
+
+void ClientLogin::setExtendedClientCap(int extended_client_cap) {
+  extended_client_cap_ = extended_client_cap;
+}
+
+void ClientLogin::setMaxPacket(int max_packet) { max_packet_ = max_packet; }
+
+void ClientLogin::setCharset(int charset) { charset_ = charset; }
+
+void ClientLogin::setUsername(std::string& username) {
+  if (username.length() <= MYSQL_MAX_USER_LEN) {
+    username_.assign(username);
+  }
+}
+
+void ClientLogin::setDb(std::string& db) { db_ = db; }
+
+void ClientLogin::setAuthResp(std::string& auth_resp) { auth_resp_.assign(auth_resp); }
+
+bool ClientLogin::isResponse41() const { return client_cap_ & MYSQL_CLIENT_CAPAB_41VS320; }
+
+bool ClientLogin::isResponse320() const { return !(client_cap_ & MYSQL_CLIENT_CAPAB_41VS320); }
+
+bool ClientLogin::isSSLRequest() const { return client_cap_ & MYSQL_CLIENT_CAPAB_SSL; }
+
+bool ClientLogin::isConnectWithDb() const { return client_cap_ & MYSQL_CLIENT_CONNECT_WITH_DB; }
+
+bool ClientLogin::isClientAuthLenClData() const {
+  return extended_client_cap_ & MYSQL_EXT_CL_PLG_AUTH_CL_DATA;
+}
+
+bool ClientLogin::isClientSecureConnection() const {
+  return extended_client_cap_ & MYSQL_EXT_CL_SECURE_CONNECTION;
+}
+
+int ClientLogin::parseMessage(Buffer::Instance& buffer, uint64_t& offset, int) {
+  if (seq_ != CHALLENGE_SEQ_NUM) {
+    return MYSQL_FAILURE;
```

Comment:
Is there a way to cover this line in the test and the similar ones (the negative path) everywhere else in this file? Basically, the code path that returns `return MYSQL_FAILURE`.